IQSS/dataverse
# only render edit files when hasValidTermsOfAccess #8742

**What this PR does / why we need it**:

One gets very strange behavior when the yellow "Info – Datasets with restricted files are required to have Request Access enabled or Terms of Access to help people access the data. Please edit the dataset to confirm Request Access or provide Terms of Access to be in compliance with the policy" message is present at the top of the dataset page.

The kebabs in the file listing on the dataset page are missing the dropdown and are non-functional.

This pull request removes the kebabs to be more in line with what we do on the file landing page. That is, now in both cases we prevent files from being edited until the dataset is in compliance. To get the dataset in compliance, you follow the instructions in the yellow info message.

**Which issue(s) this PR closes**:

- Closes #8742

**Special notes for your reviewer**:

At standup we said we'd like the dataset page to be like the file page in that we don't users editing files when the dataset is out of compliance. That's fine but the file page was only disabling the "Edit File" dropdown in CSS. It's easy to enable the button again using browser tools. So, for both the file page and the dataset page this pull request renders out the "edit file" option. More secure that way.

**Suggestions on how to test this**:

- create a dataset
- upload at least two files
- restrict one of the files (accept the defaults, which is "allow access request" check and nothing written in terms.
- publish
- Set `fileaccessrequest` to false. I only have one dataset so I ran `update termsofuseandaccess set fileaccessrequest = false;`
- As a sanity check, run `scripts/issues/8191/datasets_without_toa_or_request_access` from the 5.11 release notes to see that this is now a dataset that's out of compliance. It has no terms and it doesn't allow access requests.
- Refresh the dataset page. You should see the strange kebab with no dropdown (screenshot below).
- Try clicking the strange kebab on either file (the restricted one or the non-restricted one). The "This file has already been deleted..." message will appear the moment you click the kebab (screenshot below). Basically, no operations are allowed, not just restrict. Note also that the "Edit Files" button is disabled.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

Yes, if the dataset is out of compliance with the terms policy:

- the kebabs on the file listing page are now absent (these allow editing)
- the "Edit File" dropdown on the file page are absent (previously it was disabled)

Here are some screenshots:

![Screenshot 2022-07-27 at 16-41-30 Open Source at Harvard](https://user-images.githubusercontent.com/21006/181370874-0128e4df-d9e2-4c59-9daa-dc146224c97e.png)

![Screenshot 2022-07-27 at 16-41-43 README_private txt - Open Source at Harvard](https://user-images.githubusercontent.com/21006/181370871-16c3935c-f57d-489c-8e4f-5eb09e832765.png)

**Is there a release notes update needed for this change?**:

No.

**Additional documentation**:

None.
